src/deadlineresolver.js

```diff
@@ -208,7 +208,7 @@ class DeadlineResolver {
       debug('Resolved taskId: %s, by deadline', taskId);
 
       // Update dependency tracker
-      await this.dependencyTracker.resolveTask(taskId, 'exception');
+      await this.dependencyTracker.resolveTask(taskId, task.taskGroupId, 'exception');
```

Comment:
If we don't have a task, we won't be able to check for an empty TaskGroupActiveSet because we don't know the task-group of the non-existent task, right? I can totally see why it's nice to remove the TaskGroupActiveSet entry, but we can't really check for the emptiness, right?